### PR TITLE
UI Components into Master

### PR DESCRIPTION
**Pull request by jordanranz**
_Description of changes:_
- UI Components into modularization branch

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/amplify-ui-angular/tsconfig.legacy.json [hunk 1]
```diff
@@ -0,0 +1,8 @@
+{
```
Comment (ashika01) [inline packages/amplify-ui-angular/tsconfig.legacy.json]: Do we need this legacy config going forth?

Comment (sammartinez) [reply packages/amplify-ui-angular/tsconfig.legacy.json]: This is auto generated right?

File: packages/amplify-ui-components/LICENSE [hunk 2]
```diff
@@ -0,0 +1,21 @@
+MIT License
```
Comment (sammartinez) [inline packages/amplify-ui-components/LICENSE]: I believe we need to change this to Apache License as this is what we use in Amplify

File: packages/amplify-ui-components/package.json [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+    "@aws-amplify/auth": "^2.1.5",
+    "@aws-amplify/core": "^2.2.4",
+    "@aws-amplify/xr": "^1.1.5"
+  },
+  "husky": {
+    "hooks": {
+      "pre-push": "npm run test"
+    }
+  },
+  "license": "Apache-2.0"
```
Comment (sammartinez) [inline packages/amplify-ui-components/package.json]: interesting.

**Comment by ashika01** [comment]
@jordanranz I think we should leave out the `amplify-examples` folder out.

File: packages/amplify-ui-angular/tsconfig.legacy.json (hunk 1 above)
Comment (jordanranz) [reply packages/amplify-ui-angular/tsconfig.legacy.json]: @ericclemmons

File: packages/amplify-ui-components/LICENSE (hunk 2 above)
Comment (jordanranz) [reply packages/amplify-ui-components/LICENSE]: Changing

File: .gitignore [hunk 4]
```diff
@@ -17,3 +17,4 @@ packages/**/cjs/
 .npm/
 packages/**/cypress/videos/
 package-lock.json
+yarn.lock
```
Comment (ericclemmons) [inline .gitignore]: `yarn.lock` is removed in https://github.com/aws-amplify/amplify-js/pull/5243/files#diff-a084b794bc0759e7a6b77810e01874f2L18, so this may be out of sync.

The consensus was that `yarn.lock` shouldn't exist so that we catch regressions of transitive dependencies _sooner_, before customers do.

Having `yarn.lock` ignored may result in contributors having it locally while CI doesn't, with different behaviors.

File: package.json [hunk 5]
```diff
@@ -23,7 +23,8 @@
 		"publish:beta": "lerna publish --canary --force-publish \"*\" --yes --dist-tag=beta --preid=beta --exact",
 		"publish:release": "lerna publish --conventional-commits --yes --message 'chore(release): Publish [ci skip]'",
 		"publish:1.0-stable": "lerna publish --conventional-commits --yes --dist-tag=stable-1.0 --message 'chore(release): Publish [ci skip]'",
-		"publish:modularization": "lerna publish --canary --force-publish \"*\" --yes --dist-tag=preview --preid=preview --exact"
+		"publish:modularization": "lerna publish --canary --force-publish \"*\" --yes --dist-tag=preview --preid=preview --exact",
+		"publish:ui-components/master": "lerna publish --canary --force-publish \"*\" --dist-tag=ui-preview --preid=ui-preview --exact"
```
Comment (ericclemmons) [inline package.json]: This can be removed as part of this merge, correct?

File: .circleci/config.yml [hunk 6]
```diff
@@ -413,6 +413,7 @@ workflows:
               only:
                 - release
                 - master
+                - ui-components/master
```
Comment (ericclemmons) [inline .circleci/config.yml]: This can be removed as part of this merge, correct? (Since this branch is "done")

File: packages/amazon-cognito-identity-js/package.json [hunk 7]
```diff
@@ -1,7 +1,7 @@
 {
   "name": "amazon-cognito-identity-js",
   "description": "Amazon Cognito Identity Provider JavaScript SDK",
-  "version": "4.0.0",
+  "version": "4.1.0",
```
Comment (ericclemmons) [inline packages/amazon-cognito-identity-js/package.json]: How did this get to `4.1.0` where modularization is at `4.0.0`?
> https://github.com/aws-amplify/amplify-js/pull/5243/files#diff-d1b10c0f795c7e3e1a1e9bcc8db4b76cR4

File: packages/amplify-ui-angular/tsconfig.legacy.json (hunk 1 above)
Comment (ericclemmons) [reply packages/amplify-ui-angular/tsconfig.legacy.json]: This came from https://github.com/ionic-team/stencil-ds-plugins-demo/blob/master/packages/component-library-angular/tsconfig.legacy.json and is referenced here:
> https://github.com/ionic-team/stencil-ds-plugins-demo/blob/master/packages/component-library-angular/package.json#L16

I'm unsure to what degree we can remove this, unless we refactor it to use our internal `./scripts/build`.

File: packages/amplify-ui-components/src/components/amplify-authenticator/amplify-authenticator.tsx [hunk 8]
```diff
@@ -0,0 +1,187 @@
...
+  @Prop() federated: FederatedConfig;
+  /** Username Alias is used to setup authentication with `username`, `email` or `phone_number`  */
+  @Prop() usernameAlias: UsernameAliasStrings;
+
+  @State() authState: AuthState = AuthState.Loading;
+  @State() authData: CognitoUserInterface;
+  @State() toastMessage: string = '';
+
+  async componentWillLoad() {
+    Hub.listen(UI_AUTH_CHANNEL, data => {
```
Comment (ericclemmons) [inline packages/amplify-ui-components/src/components/amplify-authenticator/amplify-authenticator.tsx]: I'll be submitting a PR to listen to the `auth` channel. It's necessary for hosted UI redirects to complete.

File: package.json (hunk 5 above)
Comment (ashika01) [reply package.json]: I thought about this too, but if we want to test some feature development specific to UI, you could now do it in `ui-preview` before release?

File: packages/amazon-cognito-identity-js/package.json (hunk 7 above)
Comment (ashika01) [reply packages/amazon-cognito-identity-js/package.json]: could be because of all user-agent stuff that is added?

Comment (ashika01) [reply packages/amazon-cognito-identity-js/package.json]: Scratch that, you are right, it should be 4.0.0 and then the release should update it to a patch version or something

File: packages/amplify-ui-components/src/components/amplify-authenticator/amplify-authenticator.tsx (hunk 8 above)
Comment (ericclemmons) [reply packages/amplify-ui-components/src/components/amplify-authenticator/amplify-authenticator.tsx:47]: Resolved via https://github.com/aws-amplify/amplify-js/pull/5244

File: package.json (hunk 5 above)
Comment (sammartinez) [reply package.json:27]: At the moment there is no risk to keep this here, we can circle back as we need to do the same for others such as 1.0-stable command

File: .gitignore (hunk 4 above)
Comment (jordanranz) [reply .gitignore:20]: So should we remove `yarn.lock` from the gitignore?

File: .circleci/config.yml (hunk 6 above)
Comment (jordanranz) [reply .circleci/config.yml:416]: yep, I will remove these.

File: package.json (hunk 5 above)
Comment (jordanranz) [reply package.json:27]: I can remove this.

File: packages/amazon-cognito-identity-js/package.json (hunk 7 above)
Comment (jordanranz) [reply packages/amazon-cognito-identity-js/package.json:4]: yea somehow it was at 4.0.0, then the user-agent stuff was technically deployed and maybe updated this?

File: .circleci/config.yml (hunk 6 above)
Comment (ericclemmons) [reply .circleci/config.yml:594]: Follow-up PR

File: packages/auth/src/types/Auth.ts [hunk 9]
```diff
@@ -119,6 +119,7 @@ export interface FederatedResponse {
 export interface FederatedUser {
 	name: string;
 	email?: string;
+	picture?: string;
```
Comment (Amplifiyer) [inline packages/auth/src/types/Auth.ts:122]: Is this change related here?

Comment (jordanranz) [reply packages/auth/src/types/Auth.ts:122]: Yes we use `picture` in the federated UI
